conan/tools/cmake/cmake.py

```diff
@@ -133,7 +134,7 @@ def install(self, build_type=None):
         self._build(build_type=build_type, target="install")
 
     def test(self, build_type=None, target=None, output_on_failure=False):
-        if not self._conanfile.should_test:
+        if not self._conanfile.should_test or not get_env("CONAN_RUN_TESTS", True):
```

Comment:
The new conan.tools.xxxx helpers will not use env-vars for configuration, only the new [conf] mechanism.
Please use it only in the old helpers.
```suggestion
        if not self._conanfile.should_test:
```